thunder/executors/torch_autograd.py

```diff
@@ -1,22 +1,43 @@
-from dataclasses import dataclass, replace
-from functools import partial, wraps
-from inspect import signature
-from typing import Any, TYPE_CHECKING
+from dataclasses import replace
+from typing import TYPE_CHECKING
 
 import torch
 
-import thunder
 import thunder.core.utils as utils
-import thunder.distributed.prims as dist_prims
-import thunder.torch as ltorch
 from thunder.core.prims import PrimIDs
-
-from thunder.core.proxies import FutureTensorProxy, TensorProxy, variableify
-from thunder.core.pytree import tree_flatten, tree_unflatten
-from thunder.core.symbol import BoundSymbol, BoundSymbolRHS, Symbol
-from thunder.core.trace import TraceCtx
+from thunder.core.proxies import TensorProxy, variableify
+from thunder.core.pytree import tree_flatten
+from thunder.core.symbol import BoundSymbol
+from thunder.core.trace import TraceCtx, from_trace, set_tracectx, reset_tracectx
 from thunder.core.transform_common import replace_redundant_inputs
 
+if TYPE_CHECKING:
+    from thunder.core.trace import VariableInterface
+
+
+def rename_bwd_trace_outputs(bwd_trace: TraceCtx, fwd_trace: TraceCtx) -> TraceCtx:
+    trace_tok = set_tracectx(bwd_trace)
+
+    swap_map: dict[VariableInterface, TensorProxy] = {}
+    bwd_outputs, _ = tree_flatten(bwd_trace.output)
+    fwd_inputs, _ = tree_flatten((fwd_trace.args, fwd_trace.kwargs))
+
+    utils.check(len(bwd_outputs) == len(fwd_inputs), lambda: f"{len(bwd_outputs)=}, {len(fwd_inputs)=}")
+
+    for fwd_arg, bwd_out in zip(fwd_inputs, bwd_outputs):
+        if isinstance(bwd_out, TensorProxy):
+            swap_map[variableify(bwd_out)] = bwd_out.replace_name(f"grad_for_{fwd_arg.name}")
```

Comment:
If this is too verbose, it could also be 
```suggestion
            swap_map[variableify(bwd_out)] = bwd_out.replace_name(f"d{fwd_arg.name}")
```

which is somewhat common in the wild

Reply:
I'll take this with an underscore between `d` and `fwd_arg.name`

Reply:
That's unfortunate. It's strange to have tup and lst but not dct